src/Sylius/Bundle/CoreBundle/Resources/config/services/workflow/listeners.xml

```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!--
+
+ This file is part of the Sylius package.
+
+ (c) Sylius Sp. z o.o.
+
+ For the full copyright and license information, please view the LICENSE
+ file that was distributed with this source code.
+
+-->
+
+<container xmlns="http://symfony.com/schema/dic/services" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+    <services>
+        <defaults public="false" />
+
+        <service id="Sylius\Bundle\CoreBundle\EventListener\Workflow\OrderShipping\AfterOrderShippedListener">
+            <argument type="tagged_iterator" tag="sylius.workflow.callback.after_shipped_order" />
```

Comment:
As we mentioned in the internal discussion, there is not much value in the additional abstraction for defining callbacks. Let's stay as close as possible to the implementation proposed by Symfony and use for now only Workflow events.